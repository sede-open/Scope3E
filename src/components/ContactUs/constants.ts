export const TEXTAREA_MAX_LENGTH = 500;

export const FIELD_KEYS = {
  NAME: 'name',
  EMAIL: 'email',
  MESSAGE: 'message',
  ENQUIRY: 'enquiry',
  REGION: 'region',
  CONSENT: 'consent',
} as const;
