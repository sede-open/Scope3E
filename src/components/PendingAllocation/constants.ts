export enum AllocationStatus {
  MissingEmissions,
  IncomingAllocation,
  IncomingRequest,
}
