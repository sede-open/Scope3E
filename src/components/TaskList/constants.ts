import { Tasks } from './types';

export const TASK_IDS = {
  [Tasks.AREAS_OF_INTEREST]: 'areas-of-interest',
  [Tasks.BASELINE]: 'baseline',
  [Tasks.AMBITION]: 'ambition',
  [Tasks.LAST_YEAR_EMISSIONS]: 'last-year-emissions',
  [Tasks.CUSTOMER_RELATIONSHIPS]: 'customer-relationships',
  [Tasks.CUSTOMER_ALLOCATIONS]: 'customer-allocations',
  [Tasks.SUPPLIER_RELATIONSHIPS]: 'supplier-relationships',
  [Tasks.PRIVACY_SHARING]: 'privacy-sharing',
};

export enum taskListTabRoutes {
  AccountSettingsCustomer = '/account-settings/customers',
  NetworkSettingsCustomer = '/network/customers',
  ValueChainAllocateToCustomer = '/value-chain/customers',
  AccountSettingsSupplier = '/account-settings/suppliers',
  NetworkSettingsSupplier = '/network/suppliers',
  PrivacySharing = '/account-settings/your-organisation',
}
