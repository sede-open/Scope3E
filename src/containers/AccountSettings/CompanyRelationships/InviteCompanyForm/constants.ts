export const COMPANY_RELATIONSHIP_CONNECTED_ERROR =
  'You are already connected to this company';
export const COMPANY_RELATIONSHIP_PENDING_ERROR =
  'An invitation has already been sent to this this company';
export const COMPANY_RELATIONSHIP_REJECTED_ERROR =
  'An invitation has already been rejected by this company. Use the re-send option on the table list to re-invite them';
export const USER_EXISTS_ERROR = 'User not created (already exists)';
export const VETOED_COMPANY_ERROR = 'Company has been vetoed';
