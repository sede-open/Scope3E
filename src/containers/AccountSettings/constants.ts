import { NavLink } from 'components/SubheaderNavigation';

export const ACCOUNT_SETTINGS_ROUTE_PREFIX = '/account-settings';

export enum AccountSettingsRoutes {
  PersonalPreferences = '/personal-preferences',
  YourOrganisation = '/your-organisation',
  Suppliers = '/suppliers',
  Customers = '/customers',
}

export const accountSettingsNavLinks: NavLink[] = [
  {
    label: 'accountSettings:personal-preferences',
    link: AccountSettingsRoutes.PersonalPreferences,
  },
  {
    label: 'accountSettings:tabs-your-organisation',
    link: AccountSettingsRoutes.YourOrganisation,
  },
  {
    label: 'accountSettings:tabs-suppliers',
    link: AccountSettingsRoutes.Suppliers,
  },
  {
    label: 'accountSettings:tabs-customers',
    link: AccountSettingsRoutes.Customers,
  },
];
