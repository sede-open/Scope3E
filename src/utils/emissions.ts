import { DashboardDataQuery_corporateEmissions as Emission } from 'types/DashboardDataQuery';
import { SimulationDataQuery_latestCorporateEmission as LatestEmission } from 'types/SimulationDataQuery';

import { TargetStrategyType } from 'types/globalTypes';
import { round } from 'utils/number';

export const getNetEmissions = (
  emission?:
    | Emission
    | LatestEmission
    | Pick<Emission, 'scope1' | 'scope2' | 'scope3' | 'offset'>
    | null
) =>
  (emission?.scope1 ?? 0) +
  (emission?.scope2 ?? 0) +
  (emission?.scope3 ?? 0) -
  (emission?.offset ?? 0);

export const getGrossEmissions = (
  emission?: Emission | LatestEmission | null
) =>
  (emission?.scope1 ?? 0) + (emission?.scope2 ?? 0) + (emission?.scope3 ?? 0);

export type TargetChartData = {
  year?: number;
  scope1And2TargetEmission?: number;
  scope3TargetEmission?: number;
  totalTargetEmission?: number;
};

// Y = BE - ((BE - TE) / N) * X
const getModerateEmission = ({
  year: X,
  baselineEmissions: BE,
  targetEmissions: TE,
  yearPeriod: N,
}: {
  year: number;
  baselineEmissions: number;
  targetEmissions: number;
  yearPeriod: number;
}) => Math.round(BE - ((BE - TE) / N) * X);

// Y = BE - ((BE - TE) / (N^2)) * (X^2)
const getPassiveEmission = ({
  year: X,
  baselineEmissions: BE,
  targetEmissions: TE,
  yearPeriod: N,
}: {
  year: number;
  baselineEmissions: number;
  targetEmissions: number;
  yearPeriod: number;
}) => Math.round(BE - ((BE - TE) / (N * N)) * (X * X));

// Y = BE - ((BE - TE) / (sqrt(N)) * sqrt(X)
const getAggressiveEmission = ({
  year: X,
  baselineEmissions: BE,
  targetEmissions: TE,
  yearPeriod: N,
}: {
  year: number;
  baselineEmissions: number;
  targetEmissions: number;
  yearPeriod: number;
}) => Math.round(BE - ((BE - TE) / Math.sqrt(N)) * Math.sqrt(X));

const targetCalculationMethod = {
  [TargetStrategyType.AGGRESSIVE]: getAggressiveEmission,
  [TargetStrategyType.MODERATE]: getModerateEmission,
  [TargetStrategyType.PASSIVE]: getPassiveEmission,
};

type TargetDataPayload = {
  baselineData: {
    year: number;
    scope1: number;
    scope2: number;
    scope3?: number | null;
    offset?: number | null;
  };
  targetData: {
    strategy: TargetStrategyType;
    includeCarbonOffset: boolean;
    scope1And2Year: number;
    scope1And2Reduction: number;
    scope3Year?: number | null;
    scope3Reduction?: number | null;
  };
};

export const getTargetData = ({
  baselineData: { year: baselineYear, scope1, scope2, scope3 = 0, offset = 0 },
  targetData: {
    scope1And2Year,
    scope1And2Reduction,
    scope3Year,
    scope3Reduction = 0,
    strategy,
    includeCarbonOffset,
  },
}: TargetDataPayload) => {
  const getTargetEmissionData = targetCalculationMethod[strategy];

  const defaultOffset = offset ?? 0;
  const defaultScope3 = scope3 ?? 0;
  const defaultScope3Reduction = scope3Reduction ?? 0;

  const scope1And2 = scope1 + scope2;
  const scope1And2OffsetShare =
    defaultOffset * (scope1And2 / (scope1And2 + defaultScope3));
  const scope1And2BaseEmission = includeCarbonOffset
    ? scope1And2 - scope1And2OffsetShare
    : scope1And2;
  const scope1And2TargetEmission = scope1And2 * (1 - scope1And2Reduction / 100);

  const scope3OffsetShare =
    defaultScope3 === 0
      ? 0
      : defaultOffset * (defaultScope3 / (scope1And2 + defaultScope3));
  const scope3BaseEmission = includeCarbonOffset
    ? defaultScope3 - scope3OffsetShare
    : defaultScope3;
  const scope3TargetEmission =
    defaultScope3 * (1 - defaultScope3Reduction / 100);

  const scope1And2YearPeriod = scope1And2Year - baselineYear;
  const scope3YearPeriod = scope3Year ? scope3Year - baselineYear : undefined;
  const overallTargetYear =
    scope3Year && scope3Year > scope1And2Year ? scope3Year : scope1And2Year;
  const yearPeriod = overallTargetYear - baselineYear;

  const chartData: TargetChartData[] = [];
  for (let i = 0; i <= yearPeriod; i += 1) {
    const chartYear = baselineYear + i;
    // after the scope 1 and 2 year target reached,
    // for following years the emission should stay the same
    const scope1And2Emission = getTargetEmissionData({
      year: chartYear > scope1And2Year ? scope1And2YearPeriod : i,
      baselineEmissions: scope1And2BaseEmission,
      targetEmissions: scope1And2TargetEmission,
      yearPeriod: scope1And2YearPeriod,
    });

    // always fallback to the baseline values
    let scope3Emission = scope3BaseEmission ?? 0;
    if (scope3 && scope3Year && scope3YearPeriod) {
      // after the scope 3 year target reached,
      // for following years the emission should stay the same
      scope3Emission = getTargetEmissionData({
        year: chartYear > scope3Year ? scope3YearPeriod : i,
        baselineEmissions: scope3BaseEmission,
        targetEmissions: scope3TargetEmission,
        yearPeriod: scope3YearPeriod,
      });
    }

    chartData.push({
      year: chartYear,
      scope1And2TargetEmission: round(scope1And2Emission, 1),
      scope3TargetEmission: round(scope3Emission, 1),
      totalTargetEmission: round(scope3Emission + scope1And2Emission, 1),
    });
  }

  return chartData;
};

export const getTargetReductionForYear = ({
  baselineData,
  targetData,
  comparisonYear,
}: { comparisonYear: number } & TargetDataPayload) => {
  const targetEmissions = getTargetData({
    baselineData,
    targetData,
  });

  return targetEmissions.find((e) => e.year === comparisonYear);
};

export const getEmissionYears = (period: number = 16) => {
  const years: number[] = [];
  const currentYear = new Date().getFullYear();
  for (let i = 0; i < period; i += 1) {
    years.push(currentYear - i);
  }
  return years;
};

export const getYearSelectOptions = (years: number[]) =>
  years.map((year) => ({
    label: year,
    value: year,
  }));

export const getReductionPercentage = (
  currentEmissions: number,
  desiredEmissions: number
) => {
  const reductionPercentage = (1 - desiredEmissions / currentEmissions) * 100;
  const isReductionPercentagePositive = reductionPercentage > 0;
  return isReductionPercentagePositive ? reductionPercentage : 0;
};

export const getPercentileEmissions = (
  latestEmissions: number,
  percentage: number
) => Math.round(latestEmissions * (1 - percentage / 100));

export const getFutureYearOptions = (period: number = 4) => {
  const years: number[] = [];
  const currentYear = new Date().getFullYear();
  for (let i = 0; i < period; i += 1) {
    years.push(currentYear + i);
  }
  return years;
};

export const getEmissionPercentage = (total?: number, fraction?: number) => {
  if (typeof total === 'undefined' || typeof fraction === 'undefined') {
    return undefined;
  }

  return (fraction * 100) / total;
};
