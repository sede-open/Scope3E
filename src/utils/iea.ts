import {
  DEFAULT_CHART_END_YEAR,
  IEA_BELOW_2_DEGREES_SCENARIO,
  IEA_2_DEGREES_SCENARIO,
  IEA_START_YEAR,
} from '../constants';

export type IEAChartData = {
  year?: number;
  iea?: number[];
};

export const getIEAChartData = (
  emissions?: number | null,
  baselineYear?: number,
  targetYear?: number
): IEAChartData[] => {
  const chartData: { year: number; iea: number[] }[] = [];

  if (emissions && baselineYear) {
    const ieaYears: number[] = Object.keys(IEA_2_DEGREES_SCENARIO).map((e) =>
      parseInt(e, 10)
    );

    let ieaBelow2DegreeEmissions = emissions;
    let iea2DegreeEmissions = emissions;

    // want to display range starting with IEA_START_YEAR earliest
    const startYear =
      baselineYear > IEA_START_YEAR ? baselineYear : IEA_START_YEAR;

    const endYear = targetYear ?? DEFAULT_CHART_END_YEAR;

    for (let i = startYear; i <= endYear; i += 1) {
      // start year value is always 100%
      if (i === startYear) {
        chartData.push({
          year: startYear,
          iea: [iea2DegreeEmissions, ieaBelow2DegreeEmissions],
        });
      } else {
        const nextIEAYear = ieaYears.find((e) => e >= i);

        if (nextIEAYear) {
          const isIEAYear = nextIEAYear === i;

          const eiaBelow2DegreeChange =
            IEA_BELOW_2_DEGREES_SCENARIO[nextIEAYear];
          const eia2DegreeChange = IEA_2_DEGREES_SCENARIO[nextIEAYear];

          if (isIEAYear) {
            // accumulation should only happen on IEA years
            ieaBelow2DegreeEmissions = Math.round(
              ieaBelow2DegreeEmissions * eiaBelow2DegreeChange
            );
            iea2DegreeEmissions = Math.round(
              iea2DegreeEmissions * eia2DegreeChange
            );

            chartData.push({
              year: i,
              iea: [ieaBelow2DegreeEmissions, iea2DegreeEmissions],
            });
          } else {
            // calculate fractional values for years in between IEA years
            const range = Math.abs(5 - (nextIEAYear - i));

            const EIABelow2RangeValue = Math.round(
              ieaBelow2DegreeEmissions *
                (1 + (range / 5) * (eiaBelow2DegreeChange - 1))
            );
            const EIA2RangeValue = Math.round(
              iea2DegreeEmissions * (1 + (range / 5) * (eia2DegreeChange - 1))
            );

            chartData.push({
              year: i,
              iea: [EIABelow2RangeValue, EIA2RangeValue],
            });
          }
        }
      }
    }
  }

  return chartData;
};
