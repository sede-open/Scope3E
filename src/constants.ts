import { NavLink } from 'components/SubheaderNavigation';
import { Flags, getFeatureFlag } from 'utils/featureFlags';

export const mailToReport =
  'mailto:support@example.com?subject=Report%20Inappropriate%20Content%20/%20Behaviour&body=First%20Name%3A%0D%0A%0D%0A%0D%0ALast%20Name%3A%0D%0A%0D%0A%0D%0AEnquire%20About%3A';
export const mailToContactUs =
  'mailto:support@example.com?subject=Request%20to%20Terminate%20SETH%20Account%20(Changes%20to%20Terms)&body=First%20Name%3A%0D%0A%0D%0A%0D%0ALast%20Name%3A%0D%0A%0D%0A%0D%0AEnquire%20About%3A';
export const mailTo =
  'mailto:support@example.com?subject=Customer%20queries%20and%20support&body=First%20Name%3A%0D%0A%0D%0A%0D%0ALast%20Name%3A%0D%0A%0D%0A%0D%0AEnquire%20About%3A';
export const DATA_PRIVACY_NOTICE_LINK =
  'https://www.example.com/privacy/b2b-notice.html';
export const example_COOKIES_POLICY_LINK =
  'https://www.example.com/cookie-policy.html';
export const ZENDESK_SUPPLYCHAINS_HELPCENTRE_LINK =
  'https://helpcentre.supplychains.example.com/hc';

export const example_TARGET_YEAR = 2035;

export const DEFAULT_CHART_END_YEAR = 2040;

export const IEA_START_YEAR = 2014;

export const IEA_BELOW_2_DEGREES_SCENARIO: { [key: number]: number } = {
  2015: 0.981325740178918,
  2020: 0.904851880183651,
  2025: 0.894846745749108,
  2030: 0.812888691088075,
  2035: 0.774402251870676,
  2040: 0.71884044119239,
  2045: 0.655242435375798,
  2050: 0.589119947800228,
  2055: 0.382319765381074,
};

export const IEA_2_DEGREES_SCENARIO: { [key: number]: number } = {
  2015: 0.990494133381286,
  2020: 0.952014523365911,
  2025: 0.949595856516524,
  2030: 0.859579436430187,
  2035: 0.841485770804096,
  2040: 0.847733716913355,
  2045: 0.831872278050261,
  2050: 0.852399977396532,
  2055: 0.828508171439709,
};

export const EU_TARGET_REDUCTION_UP_TO_2015 = -1;
export const EU_TARGET_REDUCTION_AFTER_2015 = -1.5;

export enum PrimaryRoutes {
  Dashboard = '/dashboard',
  ValueChain = '/value-chain',
  AmbitionSimulation = '/ambition-simulation',
  Solutions = '/solutions',
  Community = '/community',
}

export const shouldDisplayCommunityPage = getFeatureFlag(
  Flags.IS_COMMUNITY_PAGE_ENABLED
);

export const primaryNavLinks: NavLink[] = [
  { label: 'common:dashboard', link: PrimaryRoutes.Dashboard },
  { label: 'common:value-chain', link: PrimaryRoutes.ValueChain },
  {
    label: 'common:ambition-simulation',
    link: PrimaryRoutes.AmbitionSimulation,
  },
  { label: 'common:solutions', link: PrimaryRoutes.Solutions },
  ...(shouldDisplayCommunityPage
    ? [{ label: 'common:community', link: PrimaryRoutes.Community }]
    : []),
];

export enum AdminDashboardRoutes {
  UsersDashboard = '/admin-dashboard/users',
  CompaniesDashboard = '/admin-dashboard/companies',
}

export const adminDashboardNavLinks: NavLink[] = [
  {
    label: 'common:users-dashboard',
    link: AdminDashboardRoutes.UsersDashboard,
  },
  {
    label: 'common:companies-dashboard',
    link: AdminDashboardRoutes.CompaniesDashboard,
  },
];

export const MAX_VERIFICATION_FILE_SIZE = 20971520;
export const VERIFICATION_FILE_MIMETYPE = 'application/pdf';

export const INPUT_DEFAULT_PLACEHOLDER = '--';

export const ONE_MILLION = 1_000_000;
export const ONE_BILLION = 1_000_000_000;
export const HUNDRED_MILLION = 100_000_000;
export const ONE_TRILLION = 1_000_000_000_000;

export const REDIRECT_SESSION_KEY = 'redirect';

export const HUBSPOT_WHITELIST = [
  'hrayr.petrosyan.web@gmail.com',
  'petrosyanhrayr95@gmail.com',
  'matthew.bridges@theagilehub.net',
  'matthew.bridges+123@theagilehub.net',
  'sitashah0607@gmail.com',
  'sita1996@hotmail.com',
  'sita1996+96@hotmail.com',
  'akanksha.dutta@theagilehub.net',
  'ryan.suarez@example.com',
];
