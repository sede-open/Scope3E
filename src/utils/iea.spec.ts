import {
  DashboardDataQuery_target as Target,
  DashboardDataQuery_corporateEmissions as Emission,
} from 'types/DashboardDataQuery';
import { DEFAULT_CHART_END_YEAR } from '../constants';
import { getNetEmissions } from './emissions';
import { getIEAChartData } from './iea';

describe('getIEAChartData()', () => {
  const baseline = {
    year: 2012,
    scope1: 10000,
    scope2: 10000,
    scope3: 24254,
    offset: 10000,
  } as Emission;
  const emissions = getNetEmissions(baseline);

  it('should return yearly iea values', () => {
    const target = {
      scope1And2Year: 2058,
    } as Target;
    const result = getIEAChartData(
      emissions,
      baseline.year,
      target.scope1And2Year
    );
    expect(result).toMatchSnapshot();
  });

  it('should return an empty array if baseline is not provided', () => {
    const result = getIEAChartData();
    expect(result).toEqual([]);
  });

  it('should return data up to DEFAULT_CHART_END_YEAR by default', () => {
    const result = getIEAChartData(emissions, baseline.year);
    expect(result[result.length - 1].year).toBe(DEFAULT_CHART_END_YEAR);
  });

  it('should return data up to target year', () => {
    const target = {
      scope1And2Year: 2030,
    } as Target;
    const result = getIEAChartData(
      emissions,
      baseline.year,
      target.scope1And2Year
    );
    expect(result[result.length - 1].year).toBe(target.scope1And2Year);
  });

  it('should return data up to 2055 if target year is later than that', () => {
    const target = {
      scope1And2Year: 2058,
    } as Target;
    const result = getIEAChartData(
      emissions,
      baseline.year,
      target.scope1And2Year
    );
    expect(result[result.length - 1].year).toBe(2055);
  });
});
