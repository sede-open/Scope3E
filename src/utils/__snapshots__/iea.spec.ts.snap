// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIEAChartData() should return yearly iea values 1`] = `
[
  {
    "iea": [
      34254,
      34254,
    ],
    "year": 2014,
  },
  {
    "iea": [
      33614,
      33928,
    ],
    "year": 2015,
  },
  {
    "iea": [
      32974,
      33602,
    ],
    "year": 2016,
  },
  {
    "iea": [
      32335,
      33277,
    ],
    "year": 2017,
  },
  {
    "iea": [
      31695,
      32951,
    ],
    "year": 2018,
  },
  {
    "iea": [
      31055,
      32626,
    ],
    "year": 2019,
  },
  {
    "iea": [
      30416,
      32300,
    ],
    "year": 2020,
  },
  {
    "iea": [
      29776,
      31974,
    ],
    "year": 2021,
  },
  {
    "iea": [
      29137,
      31649,
    ],
    "year": 2022,
  },
  {
    "iea": [
      28497,
      31323,
    ],
    "year": 2023,
  },
  {
    "iea": [
      27857,
      30998,
    ],
    "year": 2024,
  },
  {
    "iea": [
      27218,
      30672,
    ],
    "year": 2025,
  },
  {
    "iea": [
      26199,
      29811,
    ],
    "year": 2026,
  },
  {
    "iea": [
      25181,
      28949,
    ],
    "year": 2027,
  },
  {
    "iea": [
      24162,
      28088,
    ],
    "year": 2028,
  },
  {
    "iea": [
      23144,
      27226,
    ],
    "year": 2029,
  },
  {
    "iea": [
      22125,
      26365,
    ],
    "year": 2030,
  },
  {
    "iea": [
      21127,
      25529,
    ],
    "year": 2031,
  },
  {
    "iea": [
      20128,
      24693,
    ],
    "year": 2032,
  },
  {
    "iea": [
      19130,
      23857,
    ],
    "year": 2033,
  },
  {
    "iea": [
      18132,
      23022,
    ],
    "year": 2034,
  },
  {
    "iea": [
      17134,
      22186,
    ],
    "year": 2035,
  },
  {
    "iea": [
      16171,
      21510,
    ],
    "year": 2036,
  },
  {
    "iea": [
      15207,
      20835,
    ],
    "year": 2037,
  },
  {
    "iea": [
      14244,
      20159,
    ],
    "year": 2038,
  },
  {
    "iea": [
      13280,
      19483,
    ],
    "year": 2039,
  },
  {
    "iea": [
      12317,
      18808,
    ],
    "year": 2040,
  },
  {
    "iea": [
      11468,
      18176,
    ],
    "year": 2041,
  },
  {
    "iea": [
      10618,
      17543,
    ],
    "year": 2042,
  },
  {
    "iea": [
      9769,
      16911,
    ],
    "year": 2043,
  },
  {
    "iea": [
      8920,
      16278,
    ],
    "year": 2044,
  },
  {
    "iea": [
      8071,
      15646,
    ],
    "year": 2045,
  },
  {
    "iea": [
      7408,
      15184,
    ],
    "year": 2046,
  },
  {
    "iea": [
      6745,
      14722,
    ],
    "year": 2047,
  },
  {
    "iea": [
      6081,
      14260,
    ],
    "year": 2048,
  },
  {
    "iea": [
      5418,
      13799,
    ],
    "year": 2049,
  },
  {
    "iea": [
      4755,
      13337,
    ],
    "year": 2050,
  },
  {
    "iea": [
      4168,
      12880,
    ],
    "year": 2051,
  },
  {
    "iea": [
      3580,
      12422,
    ],
    "year": 2052,
  },
  {
    "iea": [
      2993,
      11965,
    ],
    "year": 2053,
  },
  {
    "iea": [
      2405,
      11507,
    ],
    "year": 2054,
  },
  {
    "iea": [
      1818,
      11050,
    ],
    "year": 2055,
  },
]
`;
